pkg/scanner/nmap.go

```diff
@@ -122,17 +122,21 @@ type NMAP struct {
 }
 
 // NewNMAP generates a Scanner implementation tha leverages NMAP internally.
-func NewNMAP(binPath string, scripts []string, scriptArgs []string) *NMAP {
-	strs := []string{
+func NewNMAP(binPath string, binArgs []string, scripts []string, scriptArgs []string) *NMAP {
+	strs := make([]string, 0, 6+len(binArgs)+len(scriptArgs))
```

Comment:
Question: Don't you need a `len(scripts)` here as well since you're appending it on line 135?

Reply:
No, it's already factored in. The scripts only constitute two elements of the slice. One element is the `--script` flag and the other is the comma joined version of the `scripts` slice.